zerver/tests/test_signup.py

```diff
@@ -1739,6 +1739,16 @@ def test_create_realm_no_creation_key(self) -> None:
             self.assertEqual(result.status_code, 200)
             self.assert_in_response('New organization creation disabled', result)
 
+    @override_settings(TERMS_OF_SERVICE=False)
+    def test_create_dev_realm(self) -> None:
+        count = UserProfile.objects.count()
+        string_id = "realm-%d" % (count,)
+        result = self.client_post('/devtools/register_realm/')
+        realm = Realm.objects.get(string_id=string_id)
+        self.assertEqual(result.status_code, 302)
+        self.assertTrue(result["Location"].startswith('http://{}.testserver'.format(string_id)))
+        self.assertEqual(realm.string_id, string_id)
+
```

Comment:
We should also verify the user is logged in as the target user using `get_session_dict_user(self.client.session)`

Reply:
I tested if the user was logged in for create new user button by checking if /login/ redirected to /
But I guess it makes sense to check if the right user was logged in.